io/io/src/TFile.cxx

```diff
@@ -9,67 +9,16 @@
  * For the list of contributors see $ROOTSYS/README/CREDITS.             *
  *************************************************************************/
 
-/**
-  \defgroup IO Input/Output Library
-
-  The library collecting the ROOT classes dedicated to data input and output.
-
-  The detailed internal description of the \ref rootio is available.
-*/
-
 /**
 \file TFile.cxx
 \class TFile
 \ingroup IO
 
-A ROOT file is a suite of consecutive data records (TKey instances) with
-a well defined format.
-
-If the key is located past the 32 bit file limit (> 2 GB) then some fields will
```

Comment:
This information is "essential" developer and advancer user information and belong in the repository (as it can and has evolved).  There should be an explicit link or reference to where in the repository that information lives.

Reply:
That's referenced as `\sa \ref rootio`, see https://root.cern/doc/master/rootio.html

If it's not in there I will add it there; let me know where you want it.

Reply:
That's a bit obscure, especially when 'just' reading the source from vi.  It probably should be clarified that the information is available/recorded(?) in `./io/doc/TFile` and in particular `tfile.md` and `tdirectory.md` 
( as a side note the 'format' in those file, both in the text version and the render version seems (to me) much harder to read than the one here (probably the missing/removed column delimiter)

Reply:
> as a side note the 'format' in those file, both in the text version and the render version seems (to me) much harder to read than the one here (probably the missing/removed column delimiter

Fine, but that shouldn't block this PR.

> That's a bit obscure, especially when 'just' reading the source from vi.

Are you complaining about doxygen syntax being obscure, or you not knowing what's in these files? (Serious question.) Either way, my mission is to deconfuse users not necessarily Philippe, and this PR seems to be a step in the right direction ;-) There are several parts missing, e.g. `TFile::Open` instead of constructor etc - but that info wasn't there to begin with. So - what do we do?

Reply:
> Fine, but that shouldn't block this PR.

In my opinion it was essential.  This documentation is not just for me but also essential for future maintainer and for external implementer of the format.  It could not be removed if there was not a (readable) alternative available.  Glad-fully that was resolved in a separate PR.

> Are you complaining about doxygen syntax being obscure, or you not knowing what's in these files? (Serious question.)

I am just stating that (possibly due to my missing in-brain doxygen parser) given those strings, I can not find the text file that contains the information.  So maybe we can __add__ outside the view of doxygen the actual name of the file or directory that contains the information (to help with browsing the code through a text editor)

Reply:
One idea that might be a good compromise. In order to not confuse users with too much technical information, but at the same time not removing all the info from the source code, one could use a dropdown menu, that would only be shown if one really wants to see more advanced info.
See https://gist.github.com/citrusui/07978f14b11adada364ff901e27c7f61

Reply:
Related to my last comment: https://github.com/doxygen/doxygen/pull/9331

Reply:
When do they tag a release with that feature?

Reply:
The ROOT online-docs are created with a self-built version of doxygen. If I remember correctly, @couet has compiled doxygen versions from git-master that were not exactly at a specific doxygen release, so maybe he can just 'git pull' and recompile on the build machine.
The next stable release will be 1.9.5 probably in four or five months I guess.

Reply:
Yes I tested 1.9.5 on my machine first before updating the one on the build machine.
It seems stable I will soon update the build machine.
And yes, we are using the most recent version of doxygen.

Reply:
Done:

```
sftnight@root-ubuntu-2004-3:~/doxygen/build$ doxygen --version
1.9.5 (fdd8e8e750e1b8e58b4d9cc758293f532278dcca)

```